x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/config/JobTests.java

```diff
@@ -594,6 +594,21 @@ public void testDocumentId() {
         assertNull(Job.extractJobIdFromDocumentId("some_other_type-foo"));
     }
 
+    public void testDeletingAndBlockReasonAreSynced() {
```

Comment:
I think we should have some BWC tests for XContent parsing. 

- We should make sure that an old job with `deleting: true` when parsed has `BlockReason.DELETE`